README.md

```diff
@@ -9,14 +9,328 @@ It provides:
+    --help="My app does something."
+    --options=[
+      cli.Option "some-option"
+        --help="This is an option."
+        --required,
+      cli.Flag "some-flag"
+        --short-name="f"
+        --help="This is a flag.",
+    ]
+    --rest=[
+      cli.Option "rest-arg"
+        --help="This is a rest argument."
+        --multi,
+    ]
+    --examples=[
+      cli.Example "Do something with the flag:"
+          --arguments="--some-option=foo --no-some-flag rest1 rest1",
+    ]
+    --run=:: | app/cli.App parsed/cli.Parsed |
+      print parsed["some-option"]
+      print parsed["some-flag"]
+      print parsed["rest-arg"]  // A list.
+      app.ui.result "Computed result"
+
+  command.run args
+```
+
+In this example, the command is called "my-app". It takes an option `some-option` and a
+flag `some-flag`. It also takes multiple rest arguments `rest-arg`.
+
+The name of the root command is used to compute the location of the cache and config
+paths. For example, if the command is called `my-app`, the cache will be stored in
+`~/.cache/my-app` and the config will be stored in `~/.config/my-app/config`. See
+the cache/config section, for more details.
+
+## Subcommands
+
+Commands can have subcommands. For example, the `git` command has subcommands like
+`git commit` and `git push`.
+
+Typically, subcommands are defined if the options/rest arguments of different actions
+are different. For example, `git commit` takes a message, but `git push` does not.
+
+Subcommands are defined by adding a `Command` object as child to another command object.
+For example:
+
+``` toit
+import cli
+
+main args/List:
+  command := cli.Command "my-app"
+    --help="My app does something."
+
+  sub := cli.Command "subcommand"
+    --help="This is a subcommand."
+    --run=:: | app/cli.App parsed/cli.Parsed |
+      print "This is a subcommand."
+  command.add sub
+
+  command.run args
+```
+
+## Options
+
+Options are parameters that are passed to the command. This package comes with
+several classes to simplify validation and conversion of common types.
+
+### Typed options
+
+Typed options are options that take a value of a specific type. For example, an
+option that takes an integer value.
+
+Here is an incomplete list of the available typed options. See the documentation
+of the library for a complete list.
+
+- `cli.OptionInt`: An option that takes an integer value. If the value is not an
+  integer, an error is returned.
+- `cli.OptionString`: An option that takes a string value. The `cli.Option` constructor
+  is an alias for this class.
+- `cli.Flag`: An option that takes a boolean value. Flags are treated specially, in
+  that they can be negated with a `no-` prefix. For example, if a flag is called
+  `some-flag`, it can be negated with `--no-some-flag`.
+- `cli.OptionEnum`: An option that takes a value from a set of allowed values.
+
+Users are encouraged to extend the `cli.Option` class and create their own typed options.
+
+## App
```

Comment:
Reading it here makes me wonder if it would be better not to abbreviate this.

Reply:
Will do it in a follow-up PR.